src/detect-engine-dns.c

```diff
@@ -119,6 +120,48 @@ int DetectEngineInspectDnsQueryName(ThreadVars *tv,
     return r;
 }
 
+int DetectEngineInspectDnsResponse(ThreadVars *tv,
+                                    DetectEngineCtx *de_ctx, DetectEngineThreadCtx *det_ctx,
+                                    const Signature *s, const SigMatchData *smd,
+                                    Flow *f, uint8_t flags, void *alstate,
+                                    void *txv, uint64_t tx_id)
+{
+    int r = 0;
+
+    SCLogDebug("start");
+
+    DNSTransaction *tx = (DNSTransaction *)txv;
+    DNSAnswerEntry *answer = NULL;
+
+    TAILQ_FOREACH(answer, &tx->answer_list, next) {
+        SCLogDebug("tx %p query %p", tx, answer);
+        if(answer->type == DNS_RECORD_TYPE_A) {
+            det_ctx->discontinue_matching = 0;
+            det_ctx->buffer_offset = 0;
+            det_ctx->inspection_recursion_counter = 0;
+
+            const uint8_t *temp = (const uint8_t *) ((uint8_t *) answer +
+                                    sizeof(DNSAnswerEntry) + answer->fqdn_len);
+
+            char a[16] = "";
+            PrintInet(AF_INET, (const void *) temp, a, sizeof(a));
+
+            uint8_t *ipstring = (uint8_t *) a;
+
+            r = DetectEngineContentInspection(de_ctx, det_ctx,
+                                              s, smd,
+                                              f, ipstring, sizeof(a), 0,
```

Comment:
seems you should pass 'strlen(a)' here, as the a is a string and we should pass the length of the actual data. Now you pass 16 even if we have '1.1.1.1' == 7